.github/workflows/docs-ci.yml

```diff
@@ -34,11 +34,22 @@ jobs:
     needs: [javadoc, typedoc, pydoc, cppdoc, rdoc]
     runs-on: ubuntu-22.04
     steps:
+      - uses: actions/checkout@v4
+      - name: Setup Git
+        run: |
+          git config user.name "$(git --no-pager log --format=format:'%an' -n 1)"
+          git config user.email "$(git --no-pager log --format=format:'%ae' -n 1)"
+          git remote remove upstream 2>/dev/null || git remote add upstream https://github.com/deephaven/deephaven-core.git
+          git fetch --tags upstream
```

Comment:
This seems pretty non-standard, fragile? I think there would be options on `actions/checkout` that accomplish what we want. For example, there is a `fetch-tags` options. Not sure why we need to change the upstream?

Reply:
This is what it took to get it to work on my fork.  For example, adding the local temp tag did not work until I added the `git config` lines.  And fetch tags didn't work without adding the upstream. For now, I wanted to at least show a pattern that works.  I'll look more into the checkout action.  Maybe attributes like fetch-tags sets some of this up automatically.

Reply:
I changed to using `fetch-tags: true` instead of the script I had.  It does not work.  I'm thinking that actions/checkout only clones from the repo it's in, which would be my fork.  It's entirely possible that this would work if Docs CI was running from deephaven-core.

Reply:
Turns out that `fetch-tags` in the checkout action is broken.  I've added a workaround suggested in the ticket: https://github.com/actions/checkout/issues/1471.  This makes the workflow slower, and it still doesn't work on a shallow fork.  I've tested the relevant parts for getting "LATEST" in a branch in the main repo and it works.